Project/DAQ_System/Program/Src/Sensor/Accelerometer/LSM6DSOXAccelerometer.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+* Accelerometer LSM6DSOX
+* Authors:       William Lim
+*
+* Email:         wll180000@utdallas.edu
+*
+* (c) 2023 Dallas Formula Racing - Embedded Firmware Team
+* Formula SAE International Collegiate Chapter
+* GPL-3.0 License
+*/
+
+#ifndef ACCELEROMETER_LSM6DSOX_H
+#define ACCELEROMETER_LSM6DSOX_H
+
+// ST HAL Dependencies
+#include "i2c.h"
+
+//  DFR Custom Dependencies
+#include "iaccelerometer.hpp"
+
+#define ACC_ADDRESS  0xd4
+
+// Register Addresses
+
+#define CTRL1_XL          0x10
+
+#define OUTX_L_A         0x28
+
+enum ODR{
+	ODR12_5,
+	ODR26,
+	ODR52,
+	ODR104,
+	ODR208,
+	ODR416,
+	ODR833,
+	ODR1_66K,
+	ODR3_33K,
+	ODR6_66K
+};
+
+enum FSR{
+	FSR2g = 10,
+	FSR4g,
+	FSR8g,
+	FSR16g
+};
```

Comment:
Instead of implicitly declaring 32-bit integers for each enum value, consider using `uint8_t`.

The [ECU library has an example](https://github.com/DallasFormulaRacing/DataAcquisition2.0/blob/develop/Project/DAQ_System/Program/Src/Sensor/ECU/PE3/frame_parsing.hpp#L26).

Reply:
Also, I have not tried this, but apparently enum classes support [implicit conversion](https://stackoverflow.com/questions/62653751/why-can-enum-class-values-of-type-int-not-be-used-as-int).

Perhaps you can assign the values from you switch statements (used in `SetODR()` and `SetFSR()`) to here, where you declare the enum values. Then, switch statements may not be needed.

Though, since it is **implicit** conversion, I wonder if this would give us a warning in our unit tests (when they are finally implemented in the future).

I will leave this part up to your discretion.